crates/miden-lib/asm/miden/note.masm

```diff
@@ -131,17 +145,35 @@ end
 #! Invocation: exec
 export.get_sender
     # pad the stack
-    padw padw padw push.0.0.0
-    # => [pad(15)]
+    padw padw padw push.0.0
+    # => [pad(14)]
 
-    exec.kernel_proc_offsets::note_get_sender_offset
-    # => [offset, pad(15)]
+    # push the flag indicating that we want to request metadata from the currently processing note
+    push.1
+    # => [is_current_note = 1, pad(14)]
+
+    exec.kernel_proc_offsets::input_note_get_metadata_offset
+    # => [offset, is_current_note = 1, pad(14)]
 
     syscall.exec_kernel_proc
-    # => [sender, pad(15)]
+    # => [METADATA, pad(12)]
+
+    # extract the sender ID from the metadata word
+    # => [aux, merged_tag_hint_payload, merged_sender_id_type_hint_tag, sender_id_prefix, pad(12)]
+
+    drop drop
+    # => [merged_sender_id_type_hint_tag, sender_id_prefix, pad(12)]
+
+    # extract suffix of sender from merged layout, which means clearing the least significant byte
+    exec.account_id::shape_suffix
+    # => [sender_id_suffix, sender_id_prefix, pad(12)]
+
+    # rearrange suffix and prefix
+    swap
+    # => [sender_id_prefix, sender_id_suffix, pad(12)]
```

Comment:
I thought we were going to move this to a stand-alone procedure? Maybe something like `extract_sender_from_metadata`.